app/src/main/kotlin/social/plasma/relay/RelayWebSocketListener.kt

```diff
@@ -0,0 +1,57 @@
+package social.plasma.relay
+
+import android.util.Log
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.runBlocking
+import okhttp3.Response
+import okhttp3.WebSocket
+import okhttp3.WebSocketListener
+import okio.ByteString
+import java.util.*
+import kotlin.coroutines.CoroutineContext
+
+class RelayWebSocketListener(
+    private val sink: MutableSharedFlow<String>,
+    private val coroutineContext: CoroutineContext
+) : WebSocketListener() {
+
+    private val subscriptionId = "plasma-sub-${UUID.randomUUID()}"
+
+    override fun onOpen(webSocket: WebSocket, response: Response) {
+        webSocket.send("""["REQ","$subscriptionId",{}]""")
+    }
+
+    /** Invoked when a text (type `0x1`) message has been received. */
+    override fun onMessage(webSocket: WebSocket, text: String) {
+        runBlocking(context = coroutineContext) { sink.emit(text) }
```

Comment:
I do not know if `runBlocking` is appropriate here, as it may interfere with androidy stuff? LMK